cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -716,11 +730,20 @@ class DeviceMemoryResource(MemoryResource):
         -------
             The shareable handle for the memory pool.
         """
-        if not self.is_ipc_enabled:
-            raise RuntimeError("Memory resource is not IPC-enabled")
-        err, alloc_handle = driver.cuMemPoolExportToShareableHandle(self._mempool_handle, _IPC_HANDLE_TYPE, 0)
-        raise_if_driver_error(err)
-        return IPCAllocationHandle._init(alloc_handle)
+        if self._alloc_handle is None:
+            if not self.is_ipc_enabled:
+                raise RuntimeError("Memory resource is not IPC-enabled")
+            if self._is_imported:
+                raise RuntimeError("Imported memory resource cannot be exported")
+            err, alloc_handle = driver.cuMemPoolExportToShareableHandle(self._mempool_handle, _IPC_HANDLE_TYPE, 0)
+            raise_if_driver_error(err)
+            try:
+                assert self._uuid is None
+                self._uuid = uuid_module.uuid4()
+                self._alloc_handle = IPCAllocationHandle._init(alloc_handle, self._uuid)
+            except:
+                os.close(alloc_handle)
```

Comment:
Do you actually intend to ignore **all** exceptions including those that inherit from `BaseException` (e.g., `KeyboardInterrupt` which is raised when the process receives a `SIGINT` signal)?